XbSymbolDatabase.h

```diff
@@ -447,6 +834,33 @@ static const char* xbe_type_str[XB_XBE_TYPE_MAX] = {
 typedef uint32_t xbaddr;
 #endif
 
+typedef struct _Library {
+    uint16_t    build_version;
+    uint16_t    qfe_version;
+    uint32_t    flag;
+    char        name[8];
+} Library;
+
+typedef struct _LibraryHeader {
+    unsigned int    count;
+    Library*        filters;
+} LibraryHeader;
+
+typedef struct _Section {
+    // These three variables must be fixed info.
+    char            name[8];
+    uint32_t        xb_virt_addr;
+    uint32_t        buffer_size;
+    void*           buffer_lower;
+} Section;
+
+typedef struct _SectionHeader {
+    unsigned int    count;
+    Section*        filters;
+} SectionHeader;
+
+typedef void* XbSymbolContext;
```

Comment:
This should be renamed to suggest that it is a handle or reference.

Current name implies a struct / class / object; so I'd expect code to work like this (which is incorrect):

```c
XbSymbolContext* ctx = malloc(sizeof(XbSymbolContext));
XbSymbol_Init(&XbSymbolContext, ...);
```

or (also incorrect):

```c
XbSymbolContext* ctx = XbSymbol_CreateContext(...);
```

In C, you usually assume a type to be an object that you don't want to assign / pass by value (as it might be too big). So if you have a non-struct-like-type, something should imply that it's a number / pointer of some sort.

Reply:
fixed